$OpenBSD: patch-libclamav_bignum_fast_h,v 1.1 2013/10/22 08:56:34 sthen Exp $

From a9d40672390a82e91a8d5c21ee2fe172fca4d2f8 Mon Sep 17 00:00:00 2001
From: David Raynor <draynor@sourcefire.com>
Date: Wed, 2 Oct 2013 15:00:23 -0400
Subject: [PATCH] bb #9017: tomsfastmath warning with zLinux on s390x

https://bugzilla.clamav.net/show_bug.cgi?id=9017

Only x86-64 is detected as a 64-bit arch which invalidates assumptions
made by the code (fp_digit must be half as long as fp_word). Better fix
would be to fix arch detection however for the purposes of clamav it's
easier to take upstream's diff verbatim.

--- libclamav/bignum_fast.h.orig	Mon Sep 16 21:28:14 2013
+++ libclamav/bignum_fast.h	Mon Oct 21 22:54:54 2013
@@ -7,6 +7,9 @@
  * 
  * Tom St Denis, tomstdenis@gmail.com
  */
+#if HAVE_CONFIG_H
+#include "clamav-config.h"
+#endif
 #ifndef TFM_H_
 #define TFM_H_
 
@@ -249,7 +252,11 @@
       typedef signed long long   long64;
    #endif
 #endif
-   typedef unsigned long      fp_digit;
+   #if (SIZEOF_LONG == 8) && (SIZEOF_LONG_LONG == 8)
+      typedef unsigned int       fp_digit;
+   #else
+      typedef unsigned long      fp_digit;
+   #endif
    typedef ulong64            fp_word;
 #endif
 
